pkg/controllers/execution/execution_controller.go

```diff
@@ -292,3 +292,12 @@ func (c *Controller) updateAppliedCondition(work *workv1alpha1.Work, status meta
 		return updateErr
 	})
 }
+
+func (c *Controller) eventf(object *unstructured.Unstructured, eventtype, reason, messageFmt string, args ...interface{}) {
+	ref, err := helper.GenEventRef(object)
+	if err != nil {
+		klog.Errorf("ignored event(%s) as failed to build event reference to: '%#v' due to %v", reason, object, err)
```

Comment:
```suggestion
		klog.Errorf("ignore event(%s) as failed to build event reference for: kind=%s, %s/%s due to %v", reason, object.GetKind(), object.GetNamespace(), object.GetName(), err)
```

log the whole object might be verbose.